client/rpc/src/eth/execute.rs

```diff
@@ -112,9 +112,9 @@ where
 			}
 		};
 
-		if let Ok(BlockStatus::Unknown) = self.client.status(id) {
+		let Ok(_hash) = self.client.expect_block_hash_from_id(&id) else {
```

Comment:
Perhaps being pedantic, but IMO the use of `let else` deviates from the intent of the check, where we explicitly test for the presence of the block error, instead of the current intent of using the result and handling if an error occurs.


A more straightforward fix would be to explicitly test for `UnknownBlock`:

```rust
if let Err(sp_blockchain::Error::UnknownBlock(_)) = self.client.expect_block_hash_from_id(&id) {
	return Err(crate::err(JSON_RPC_ERROR_DEFAULT, "header not found", None));
}
```